xbmc/pvr/addons/PVRClient.cpp

```diff
@@ -1379,79 +1379,71 @@ std::shared_ptr<CPVRClientMenuHooks> CPVRClient::GetMenuHooks()
 PVR_ERROR CPVRClient::CallEpgTagMenuHook(const CPVRClientMenuHook& hook, const std::shared_ptr<CPVREpgInfoTag>& tag)
 {
   return DoAddonCall(__FUNCTION__, [&hook, &tag](const AddonInstance* addon) {
-    PVR_MENUHOOK_DATA hookData = {};
-    hookData.cat = PVR_MENUHOOK_EPG;
-    hookData.data.iEpgUid = tag->UniqueBroadcastID();
+    CAddonEpgTag addonTag(tag);
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = PVR_MENUHOOK_EPG;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallEPGMenuHook(&menuHook, &addonTag);
   });
 }
 
 PVR_ERROR CPVRClient::CallChannelMenuHook(const CPVRClientMenuHook& hook, const std::shared_ptr<CPVRChannel>& channel)
 {
   return DoAddonCall(__FUNCTION__, [&hook, &channel](const AddonInstance* addon) {
-    PVR_MENUHOOK_DATA hookData = {};
-    hookData.cat = PVR_MENUHOOK_CHANNEL;
-    WriteClientChannelInfo(channel, hookData.data.channel);
+    PVR_CHANNEL tag;
+    WriteClientChannelInfo(channel, tag);
 
-    PVR_MENUHOOK menuHook = {0};
-    menuHook.category = hookData.cat;
+    PVR_MENUHOOK menuHook;
+    menuHook.category = PVR_MENUHOOK_CHANNEL;
     menuHook.iHookId = hook.GetId();
     menuHook.iLocalizedStringId = hook.GetLabelId();
 
-    return addon->toAddon->MenuHook(menuHook, hookData);
+    return addon->toAddon->CallChannelMenuHook(&menuHook, &tag);
```

Comment:
Why do I have to set `menuHook.category`? The called method already defines the semantics, so no need to say which "category". It's clear that it must be "channel" when calling `CallChannelMenuHook`. Same comment applies to the other "Call*MenuHook" calls.